centaur/src/main/resources/standardTestCases/retry_with_more_memory/do_not_retry_rc0.wdl

```diff
@@ -0,0 +1,21 @@
+version 1.0
+
+task imitate_oom_error {
+  command {
+    # This task mimics printing an exception that would cause it to retry with increased memory, but
+    # because the return code is 0 (success) the task does not retry.
+    printf "Exception in thread "main" java.lang.OutOfMemoryError: testing\n\tat Test.main(Test.java:1)\n" >&2 && (exit 0)
```

Comment:
You shouldn't need the `exit 0` because the printf will presumably have a 0 return code anyway, and the script is about to end.

I _think_ Cromwell will catch attempts by users to `exit` in their scripts and perform the required cromwell tidy-up afterwards anyway, but just in case, I would omit it here.

Reply:
Done. And I figured out the syntax for referencing the metadata. I still confused why it is so different for the cases where it retries though--

`retry_same_memory_output_failure.imitate_oom_error.-1.1.executionStatus`
vs.
`calls.do_not_retry_rc1.imitate_oom_error.executionStatus`

Reply:
Done… and I figured out the syntax for checking the metadata. I wrote a comment about this right before standup, but somehow it disappeared.

Reply:
-1 refers to the shard index / scatter index of the call. Inside call metadata, there is a field called `shardIndex`, and if that call doesn't have any scatters, the shardIndex would be -1. AFAIK, the metadata assertion key structure in Centaur is flattened structure of actual metadata, hence it includes `<shardIndex>.<attempt>` in the assertion key to uniquely identify call metadata.

Reply:
Thanks @salonishah11. It's odd though because I couldn't get the syntax with the -1 shard to work in the case where the task succeeds on the first run. The only thing I could get to work was `calls.do_not_retry_rc0.imitate_oom_error.executionStatus` (which does match the metadata structure).